XamarinCommunityToolkit/Views/GravatarImageSource/MacOS/GravatarImageSourceHandler.macos.cs

```diff
@@ -0,0 +1,33 @@
+using System.Threading;
+using System.Threading.Tasks;
+using AppKit;
+using Foundation;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.MacOS;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class GravatarImageSourceHandler : IImageSourceHandler
+	{
+		public async Task<NSImage> LoadImageAsync(ImageSource imagesource, CancellationToken cancelationToken = default, float scale = 1)
+		{
+			var fileInfo = await GravatarHandlerUtil.Load(imagesource, 1, GetCacheDirectory()).ConfigureAwait(false);
+
+			if (fileInfo?.Exists ?? false)
+				return new NSImage(fileInfo.FullName);
+
+			return null;
+		}
+
+		static string GetCacheDirectory()
+		{
+			var dirs = NSSearchPath.GetDirectories(NSSearchPathDirectory.CachesDirectory, NSSearchPathDomain.User);
+			if (dirs == null || dirs.Length == 0)
+			{
+				// this should never happen...
```

Comment:
Do we need this comment? If this should never happen, can be a good idea trown an exception?

Reply:
this is the literal copy from Xamarin.Essentials since there is no dependency on Essentials